apis/oauth/src/test/java/org/jclouds/oauth/v2/AuthorizationApiAzureLiveTest.java

```diff
@@ -0,0 +1,88 @@
+
+import org.jclouds.apis.BaseApiLiveTest;
+import org.jclouds.oauth.v2.config.OAuthModule;
+import org.jclouds.oauth.v2.config.OAuthProperties;
+import org.jclouds.oauth.v2.config.CredentialType;
+import org.jclouds.oauth.v2.config.OAuthScopes;
+import org.jclouds.oauth.v2.config.OAuthScopes.SingleScope;
+import org.jclouds.oauth.v2.domain.Token;
+import org.jclouds.providers.ProviderMetadata;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.name.Names;
+
+@Test(groups = "live", singleThreaded = true)
+public class AuthorizationApiAzureLiveTest extends BaseApiLiveTest<AuthorizationApi> {
+
+    private String scope;
+    private String resource;
+    private String audience;
+
+    public AuthorizationApiAzureLiveTest() {
+        provider = "oauth";
+    }
+
+    public void authenticateCredentialAndPassword() throws Exception {
+        Token token = api.authorizeClientSecret(identity, credential, resource);
+
+        assertNotNull(token, "no token when authorizing password based credential");
+    }
+
+    /** OAuth isn't registered as a provider intentionally, so we fake one. */
+    @Override protected ProviderMetadata createProviderMetadata() {
+        return forApiOnEndpoint(AuthorizationApi.class, endpoint).toBuilder().id("oauth").build();
+    }
+
+    @Override protected Properties setupProperties() {
+        Properties props = super.setupProperties();
+        props.put(CREDENTIAL_TYPE, CredentialType.CLIENT_CREDENTIALS_SECRET.toString());
+        resource = "https://management.azure.com/";
+        props.put(RESOURCE, resource);
+        endpoint = checkNotNull(setIfTestSystemPropertyPresent(props, OAuthProperties.AZURE_ENDPOINT), OAuthProperties.AZURE_ENDPOINT);
+        identity = checkNotNull(setIfTestSystemPropertyPresent(props, OAuthProperties.AZURE_IDENTITY), OAuthProperties.AZURE_IDENTITY);
+        credential = checkNotNull(setIfTestSystemPropertyPresent(props, OAuthProperties.AZURE_CREDENTIAL), OAuthProperties.AZURE_CREDENTIAL);
+        audience = checkNotNull(setIfTestSystemPropertyPresent(props, OAuthProperties.AUDIENCE), "test.jclouds.oauth.audience");
+        scope = resource;
+        return props;
+    }
+
+    @Override protected Iterable<Module> setupModules() {
+        return ImmutableList.<Module>builder() //
+                .add(new OAuthModule()) //
+                .add(new Module() {
+                    @Override public void configure(Binder binder) {
+                        // ContextBuilder erases oauth.endpoint, as that's the same name as the provider key.
+                        binder.bindConstant().annotatedWith(Names.named("oauth.endpoint")).to(endpoint);
+                        binder.bind(OAuthScopes.class).toInstance(SingleScope.create(scope));
+                    }
+                }).addAll(super.setupModules()).build();
+    }
+}
```

Comment:
If possible, get rid of this class and just add the new method to the existing live test.